### PR TITLE
Add annual report for SIG-Storage

### LINKED ISSUE #5513: 2021 Annual Report: SIG Storage
Chairs: @saad-ali @xing-yang 
Liasion: @parispittman

- [ ] Consult SIG to complete draft of report
- [ ] If needed, consult with Steering Committee liasion on any private issues or concerns that arrise while completing report
- [ ] Submit PR to git.k8s.io/community repo with copy of report, assigning your SC liasion for review
- [ ] Steering Committee reviews report, and may make comments/ask questions
- [ ] Once all comments are addressed, SC will approve report to merge
- [ ] If needed, any follow up items may be brought to the "Chairs, Tech Leads, and Organizers" meeting in April

Once all the above items are complete, this issue may be `/close`'d

Key dates:
- Initial PR to communtiy repo should be opened by March 1st, 2021
- PR should be reviewed and merged by April 1st, 2021

More detailed information on the annual reports process is available [here](https://git.k8s.io/community/committee-steering/governance/annual-reports.md).

If you have any questions or concerns about this process, you may reach the Steering Committee via the following methods:
- Slack/E-mail your liasion directly
- The public #steering-committee channel in Slack
- The public mailing list steering@kubernetes.io
- The private mailing list steering-private@kubernetes.io for any private or sensitive issues.

/sig storage
/committee steering

### PR DESCRIPTION
**Pull request by xing-yang**
**Which issue(s) this PR fixes**:
Fixes #5513

CC @saad-ali @msau42 @jsafrane

### CODE REVIEW

File: sig-storage/annual-report-2021.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  * COSI: https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support
+  * Non-graceful node shutdown: https://github.com/kubernetes/enhancements/pull/1116
+  * Allow Kubernetes to supply pod's fsgroup to CSI driver on mount: https://github.com/kubernetes/enhancements/pull/2323
+  * Volume group: https://github.com/kubernetes/enhancements/pull/1551
+  * Recover from resize failures: https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1790-recover-resize-failure
+  * SELinux recursive permission handling: https://github.com/kubernetes/enhancements/issues/1710
+  * Volume/Snapshot namespace transfer: https://github.com/kubernetes/enhancements/pull/2326
+  * Volume health (2nd alpha due to re-design): https://github.com/kubernetes/enhancements/pull/2286
+  * Prioritization on volume capacity (sig-scheduling): https://github.com/kubernetes/enhancements/pull/1862
+  * PVC created by statefulset will not be auto removed (sig-apps): https://github.com/kubernetes/enhancements/pull/1915
```
Comment (mattcary) [inline sig-storage/annual-report-2021.md:130]: Maybe better to point at the submitted KEP for PVC auto-removal? https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/1847-autoremove-statefulset-pvcs

File: sig-storage/annual-report-2021.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+ContainerNotifier (sig-node): https://github.com/kubernetes/enhancements/pull/1995
+  * User id ownership in configmaps and secrets (sig-auth)
+
+What areas and/or subprojects does the group need the most help with?
+* There are features that we co-owned with sig-node, sig-apps, etc. We would need help from other SIGs to move things forward.
+  * Features co-owned with sig-node:
+    * ContainerNotifier: https://github.com/kubernetes/enhancements/pull/1995
+    * Volume health: https://github.com/kubernetes/enhancements/pull/2286
+  * Features co-owned with sig-apps:
+    * PVC created by statefulset will not be auto removed: https://github.com/kubernetes/enhancements/pull/1915
```
Comment (mattcary) [inline sig-storage/annual-report-2021.md:142]: and here. https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/1847-autoremove-statefulset-pvcs

**Review by mattcary — COMMENTED** [review]
Thanks for pulling this together!

File: sig-storage/annual-report-2021.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+85% open to lgtm: avg 1.4 week, max 14.11 week
+85% lgtm to approve: avg 17.11 hour, max 3.92 week
+85% approve to merge: avg 1.2 day, max 1.86 week
+https://k8s.devstats.cncf.io/d/44/pr-time-to-approve-and-merge?orgId=1&var-period=d7&var-repogroup_name=SIG%20Storage&var-apichange=All&var-size_name=All&var-kind_name=All
+
+Age of 7 day MA of issues by sig-storage repository group on 1/25/2021:
+Median time to close issue: Min 0.93 hour, Max 23.55 week, Avg 3.30 week
+https://k8s.devstats.cncf.io/d/15/issues-age-by-sig-and-repository-groups?orgId=1&var-period=d7&var-repogroup_name=SIG%20Storage&var-sig_name=All&var-kind_name=All&var-prio_name=All
+
+Suggestions on how to improve the PR/Issue velocity will be beneficial.
```
Comment (msau42) [inline sig-storage/annual-report-2021.md:164]: Some suggestions I have:
- increase the reviewer pool
- regular issue triage meetings

File: sig-storage/annual-report-2021.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    * Volume expansion for statefulset: https://github.com/kubernetes/enhancements/pull/1848
+  * Features co-owned with sig-scheduling:
+    * Prioritization on volume capacity (sig-scheduling): https://github.com/kubernetes/enhancements/pull/1862
+
+* There are also areas where we need sig-architecture’s help. For example, VolumeSnapshot is already GA and it is a core feature in sig-storage developed using CRDs. We have documented and blogged that components such as the snapshot CRDs and snapshot controller need to be deployed by Kubernetes distros, however, not every distro is installing them. This is a problem with any feature developed using CRD. We hope sig-architecture can help us resolve this problem.
+
+What's the average open days of a PR and Issue in your group? / what metrics does your group care about and/or measure?
+
+According to the following stats on 1/25/2021, it shows the 7 day MA for a PR Time to Approve and Merge in sig-storage repository group is:
+Open to lgtm: avg 3.55 days, max 14.11 week
```
Comment (nikhita) [inline sig-storage/annual-report-2021.md:152]: nit: might be good to add this as bullet points because it isn't formatted well in the rendered markdown

**Review by nikhita — COMMENTED** [review]
I really, _really_ love this!! Lgtm with steering hat on. :tada:

File: sig-storage/annual-report-2021.md (hunk 1 above)
Comment (xing-yang) [reply sig-storage/annual-report-2021.md:130]: Done

File: sig-storage/annual-report-2021.md (hunk 2 above)
Comment (xing-yang) [reply sig-storage/annual-report-2021.md:142]: Done

File: sig-storage/annual-report-2021.md (hunk 3 above)
Comment (xing-yang) [reply sig-storage/annual-report-2021.md:164]: Added.

File: sig-storage/annual-report-2021.md (hunk 4 above)
Comment (xing-yang) [reply sig-storage/annual-report-2021.md]: Fixed.

**Comment by xing-yang** [comment]
Addressed review comments.  Thanks.

**Comment by parispittman** [comment]
/unhold

leaving lgtm to @msau42 @saad-ali @jsafrane 

tysm!

**Pull request merged**